pubcid.js/src/index.js

```diff
@@ -2,14 +2,25 @@ const SOURCE = 'pubcid.org';
 
 window.googletag = window.googletag || {cmd: []};
 window.googletag.encryptedSignalProviders = window.googletag.encryptedSignalProviders || [];
+window.pbjs =  window.pbjs || [];
 
 function collector() {
-    let value = getCookie();
-    if(value){
-        return;
-    }
-    value = generateUUID();
-    document.cookie = SOURCE + '=' + value;
+    return new Promise((resolve, reject) => {
+        let cookieValue = getCookie();
```

Comment:
why get the cookie first if the eids is populated? Why not just go to eids first?

Reply:
@jdwieland8282  Need your inputs here. Should we reset the value even if the cookie value already exists?

Reply:
If the sharedid in eids and cookies could be different we should preference the one we think is oldest.  If we think the two values will always be the same we should read from whichever is fastest.

Reply:
@jdwieland8282 The id value doesn't contain any date info hence we cant identify which ids are older

Reply:
I think we should just always prefer the eid, publishers would likely be surprised if a different value was ever pulled. One other thing to note, this script uses cookies, but publishers often use local storage. Publishers might not like it if they deployed this script and saw their first party cookie grow, potentially hitting cdn limits they've carefully just avoided.

Reply:
ok lets use `eids` then (unless missing), lets also write the sharedid to LS

Reply:
Have updated the script to read from Eids and store in LS